src/v/pandaproxy/server.cc

```diff
@@ -137,7 +139,7 @@ void server::route(server::route_t r) {
     // NOTE: this pointer will be owned by data member _routes of
     // ss::httpd:server. seastar didn't use any unique ptr to express that.
     auto* handler = new handler_adaptor(
-      _pending_reqs, _ctx, std::move(r.handler), r.path_desc);
+      _pending_reqs, _ctx, std::move(r.handler), r.path_desc, _server_name);
```

Comment:
For the rest proxy we get "pandaproxy", It might be preferable to make it "rest_proxy". I'm not sure of the implication of renaming the server - I think it also affects a field in the rpc metrics, which may or may not be desired.

Reply:
By changing the server name, the older metrics endpoint would refer to pandaproxy under both names (pandaproxy and rest_proxy). This is because the `server` object metrics use the server name as a label. I think that would be confusing.

I feel like I'm missing some context here in order to make a decision. In general, do we want to rename `pandaproxy` to `rest_proxy`?